OS/DiskOS/Editors/code.lua

```diff
@@ -219,16 +221,46 @@ function ce:drawLineNum()
   color(eapi.flavorBack) print(linestr,1, self.sh-self.fh-2)
 end
 
+function ce:searchTextAndNavigate(from_line)
+ -- cprint("simple test dev mode")
+ for i,t in ipairs(buffer)
+ do
+  if from_line~=nil and i<= from_line then
+   cprint("we pass ")
+   cprint(i)
+  else
+   if string.find(t,ce.searchtxt) then
+    cprint("txt found at line")
+    cprint(i)
+    self.cy=i
+    ce:checkPos()
+    ce:drawBuffer()
+    break
+   end
+  end
+ end
+
+end
+
 function ce:textinput(t)
-  self:beginUndoable()
-  local delsel
-  if self.sxs then self:deleteSelection(); delsel = true end
-  buffer[self.cy] = buffer[self.cy]:sub(0,self.cx-1)..t..buffer[self.cy]:sub(self.cx,-1)
-  self.cx = self.cx + t:len()
-  self:resetCursorBlink()
-  if self:checkPos() or delsel then self:drawBuffer() else self:drawLine() end
-  self:drawLineNum()
-  self:endUndoable()
+  cprint(t)
+  if ce.incsearch==true then
```

Comment:
Hey !, there's nothing called `variable == true` in Lua !, instead do:
```lua
if ce.insearch then
 --..
end
```
Another thing, since we are naming it `ce:searchTextAndNavigate` then you better do
```lua
if self.incsearch then
 --..
end
```

Reply:
you so in this context "self" is the object/table that contains the function?